internal/context.go

```diff
@@ -486,7 +486,7 @@ func (c *valueCtx) Value(key interface{}) interface{} {
 	return c.Context.Value(key)
 }
 
-func spanFromContext(ctx Context) opentracing.SpanContext {
+func GetSpanContext(ctx Context) opentracing.SpanContext {
```

Comment:
should we also expose `contextWithSpan` or is there another way for users to add their existing context?

Reply:
Now I'm a bit concerned about the determinism if users start a new span inside the workflow.

Reply:
I guess we don't have a clear story around replay traces. It wouldn't cause non-determinism though. Just weird/duplicate traces potentially. 
Are you suggesting to expose the span via `GetSpanContext` but don't let user set/override the span in the context?

Reply:
You are right. It's still deterministics. Replay is going to cause issues. We will see multiple child spans if we allow users to set them.

Reply:
Using Sideeffect would work. But is span serializable or recoverable?

Reply:
SpanContext is serializable and should be passed. With that said, we'll require users to start a short-lived new span just to modify spancontext. Please refer the examples in the comments

Reply:
I've added an integration test which works.